src/meta/src/controller/system_param.rs

```diff
@@ -0,0 +1,316 @@
+// Copyright 2023 RisingWave Labs
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use std::sync::Arc;
+use std::time::Duration;
+
+use anyhow::anyhow;
+use risingwave_common::system_param::reader::SystemParamsReader;
+use risingwave_common::system_param::{
+    check_missing_params, derive_missing_fields, set_system_param,
+};
+use risingwave_common::{for_all_params, key_of};
+use risingwave_pb::meta::subscribe_response::{Info, Operation};
+use risingwave_pb::meta::PbSystemParams;
+use sea_orm::{ActiveModelTrait, ActiveValue, DatabaseConnection, EntityTrait, TransactionTrait};
+use tokio::sync::oneshot::Sender;
+use tokio::sync::RwLock;
+use tokio::task::JoinHandle;
+use tracing::info;
+
+use crate::controller::SqlMetaStore;
+use crate::manager::{LocalNotification, NotificationManagerRef};
+use crate::model_v2::prelude::SystemParameter;
+use crate::model_v2::system_parameter;
+use crate::{MetaError, MetaResult};
+
+pub type SystemParamsControllerRef = Arc<SystemParamsController>;
+
+pub struct SystemParamsController {
```

Comment:
May I understand `Controller` as the next version of `Manager`?

Reply:
Yes exactly, I was trying to follow MVC model. 🥵